core/state_processor.go

```diff
@@ -95,9 +103,6 @@ func getTransactionType(header *types.Header, tx *types.Transaction) types.Trans
 	if tx.ShardID() != tx.ToShardID() && header.ShardID == tx.ShardID() {
 		return types.SubtractionOnly
 	}
-	if tx.ShardID() != tx.ToShardID() && header.ShardID == tx.ToShardID() {
```

Comment:
Why only this is deleted?

Reply:
we don't have the additiononly type transaction; use receipts to add balance